pre-sync/oci-image-verification/README.md

```diff
@@ -0,0 +1,188 @@
+# Sample OCI Image Signature Verification with Config Sync
+
+This sample demonstrates how to verify the signature of an OCI image that Config
+Sync is managing. The webhook monitors the configsync.gke.io/image-to-sync
+annotation on specified RootSync or RepoSync resources, which contains the full
+URL of the image to sync. By comparing the previous and updated values in the
+admission review request, the webhook can detect if a new image has been
+introduced, triggering an image verification process.
+
+For one way of authenticating Cosign client to the source image registry, this
+sample uses the `cosign login` command with a token stored in a Kubernetes
+secret. Token can be expired, user could also build a custom authentication mechanism.
+
+For another example tailored for Cosign library integration, please see the
+[example] in the Config Sync repository.
+
+## Prerequisites
+
+CLI: [OpenSSL], [Cosign], [Gcloud], [Docker], [Kubectl], [Crane]
+
+Environment: GKE cluster, Google Cloud Artifact Registry repo
+
+## Build the signature verification server
+
+```shell
+docker build -t <IMAGE_REGISTRY_URL>:latest . && docker push <IMAGE_REGISTRY_URL>:latest
+```
+
+## Create a Namespace and Kubernetes Service Account
+
+This is intended to group everything under the same namespace. In this example, `signature-verification` is used as a reference:
+
+```shell
+kubectl create ns signature-verification
+```
+
+It is recommended to create the Kubernetes Service Account in advance to facilitate authentication in subsequent steps:
+
+```shell
+kubectl create sa signature-verification-sa -n signature-verification
+```
+
+## Authentications
+
+### Image registry authentication
+
+In this example, a token is used to authorize Cosign with the Docker registry.
+The token is stored in a Kubernetes secret within the cluster, and the server
+will not start if the token is missing. Note that the token has an expiration
+time, so for production environments, a more robust authentication method for
+the image verification client is recommended. For an example using the Cosign
+library with built-in authentication, see this [example].
+
+```shell
+TOKEN=$(gcloud auth print-access-token)
+
+kubectl create secret generic registry-token \
+  --namespace signature-verification \
+  --from-literal=token=$TOKEN
+```
+
+### IAM setup for the signature verificiation server
+
+- Give the Google service account permission to read images from Artifact Registry
```

Comment:
Instead of impersonating a GSA, have you tried granting KSA (BYOID) the reader role directly?

Reply:
Tested using BYOID to grant access to KSA works, modified the README.